ministryofjustice/correspondence_tool_staff
# CT-2368 closed cases report for offender sar

## Description

There has been a closed report template for all case types including FOI, SAR, etc. but the problem with that template is that there will be some columns meaningful to one of case type but not irrelevant to other case types, it becomes worse for Offender SAR type,  this PR is to 
- create a close report for Offender SAR specific
- have the flexibility to allow adding another close report easily in the future 
- solve the report categories permission issues to show up the reports which is allowed to be seen by the current user  

## Self-review checklist


### Screenshots

- for other type
<img width="665" alt="Screenshot 2020-07-30 at 17 27 20" src="https://user-images.githubusercontent.com/26111671/88948600-089c4400-d28a-11ea-8036-fba2c97aceaf.png">
<img width="464" alt="Screenshot 2020-07-30 at 17 27 30" src="https://user-images.githubusercontent.com/26111671/88948619-0e922500-d28a-11ea-8d01-35abbd962984.png">

- for offender sar
<img width="703" alt="Screenshot 2020-07-30 at 17 27 44" src="https://user-images.githubusercontent.com/26111671/88948667-16ea6000-d28a-11ea-8c56-275aa0699c0d.png">
<img width="523" alt="Screenshot 2020-07-30 at 17 27 54" src="https://user-images.githubusercontent.com/26111671/88948673-1a7de700-d28a-11ea-8a11-784a8e4477b7.png">


### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-2368
### Deployment


### Manual testing instructions

1 - login as branson user
2 - choose reports tabl
3 - click "Create report " button
4 - Choose "Offender SAR" and the sub report of "Closed cases report"
5 - choose the time frame
9 - confirm to  "Create custom report"
10 - click the link to download it from flash message
